src/modules/hip/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,411 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 -  moving mean square kernel device helpers --------------------
+
+// calculate the position in shared memory to avoid bank conflicts
+__host__ __device__ __forceinline__ int compute_pos_in_smem(int pos)
+{
+    return pos + (pos >> 5); // since shared memory banks considered is 32
+}
+
+__device__ __forceinline__ void compute_prefix_sum(float *input, uint bufferLength)
```

Comment:
This device helper function could use some comments

Reply:
Done
Somehow it is not getting reflected in github UI
https://github.com/sampath1117/rpp/blob/sr/nsr_hip/src/modules/hip/kernel/non_silent_region_detection.hpp#L14